apps/src/sites/studio/pages/datasets/show_manifest.js

```diff
@@ -18,5 +18,15 @@ $(document).ready(function() {
     </Provider>,
     document.querySelector('.manifest_editor')
   );
-  initializeCodeMirror('content', 'application/json');
+  initializeCodeMirror('content', 'application/json', {callback: onChange});
 });
+
+function onChange(editor) {
+  try {
+    const newManifest = JSON.parse(editor.getValue());
+    getStore().dispatch(setLibraryManifest(newManifest));
+  } catch (e) {
+    // There is a JSON error. Set the manifest to {} so that we show an error instead of the library preview
+    getStore().dispatch(setLibraryManifest({}));
```

Comment:
Here and other places, can you change `libraryManifest` to `dataLibraryManifest` or something similar? `library` is starting to become an overloaded term in our codebase.

Reply:
Addressed on https://github.com/code-dot-org/code-dot-org/pull/33061